### ISSUE TITLE
SSH timeout not configurable enough

### ISSUE THREAD

**Issue by GreenMobius** [pexpect/pexpect#701]
I've been having an issue recently where I have to wait for an extremely long time for my ssh sessions to complete logging in. The time delay happens to be 30 seconds, which is the default for pexpect's expect method. Doing a little digging led me to the below snippet of code, where I believe the following is happening:
- First phase receives a password prompt back
- Second phase times out, unable to detect the login prompt

The shell prompt looks like `<user>@<host>:~#`, so with the `r"[#$]"` defined as `original_prompt` (which I have not modified), the prompt SHOULD be detected. Since it's not, I get forced into a timeout that's much longer than I expected. Help on this would be appreciated, but my main issue here is the length of this timeout.

If I'm able to specify a `login_timeout` when logging into my `pxssh` session, shouldn't that timeout be applied to all of the follow-up phase expects as well? I know that I'm able to modify the default time via the `pexpect.spawn` constructor, but that applies it to all of my expects.

tl;dr Add `timeout=login_timeout` to all the expects in the phase 1 switch.

Excerpt from `pxssh.py` for reference
```python
        # This does not distinguish between a remote server 'password' prompt
        # and a local ssh 'passphrase' prompt (for unlocking a private key).
        i = self.expect(session_init_regex_array, timeout=login_timeout)

        # First phase
        if i==0:
            # New certificate -- always accept it.
            # This is what you get if SSH does not have the remote host's
            # public key stored in the 'known_hosts' cache.
            self.sendline("yes")
            i = self.expect(session_regex_array)
        if i==2: # password or passphrase
            self.sendline(password)
            i = self.expect(session_regex_array)
        if i==4:
            self.sendline(terminal_type)
            i = self.expect(session_regex_array)

...

        elif i==5: # Timeout
            # <comments>
            pass
```

My usage looks like the following
```python
child = pexpect.pxssh.pxssh()
pex.login(host, user, pass, login_timeout=5)
pex.prompt()
...
```

**Comment by Red-M** [comment]
This sounds like the remote side is under extreme load and isn't able to reply in a timely fashion, I think you should try extending the timeout overall and seeing how that goes.

**Comment by Red-M** [comment]
A small reminder that pxssh is not recommend to be used as per [the documentation](https://pexpect.readthedocs.io/en/stable/api/pxssh.html#pxssh-control-an-ssh-session).

Please re-open this if you think this is still an issue.
